### PR TITLE
Allow install on Symfony3

### PR DESCRIPTION
**Pull request by mrardon**
Not sure if this is the most "correct" way of allowing this but wanted to give it a shot.

---

This change is [Reviewable](https://reviewable.io/reviews/michelsalib/bccresquebundle/136)

### CODE REVIEW

File: composer.json [hunk 1]
```diff
@@ -13,7 +13,7 @@
         }
     ],
     "require": {
-        "symfony/framework-bundle": ">=2.0,<3.0",
+        "symfony/framework-bundle": ">=2.0,>=3.0",
```
Comment (lchristian13) [inline composer.json]: Please change to ">=2.0,<=4.0" to all compatibility with Symfony 2. Like this it is allowed to be used only with Symfony 3.

**Comment by mrardon** [comment]
Changes made to allow Symfony 2 and 3

File: composer.json [hunk 2]
```diff
@@ -13,7 +13,7 @@
         }
     ],
     "require": {
-        "symfony/framework-bundle": ">=2.0,<3.0",
+        "symfony/framework-bundle": ">=2.0,<=4.0",
```
Comment (reminec) [inline composer.json]:
``` yaml
"symfony/framework-bundle": "^2.0|~3.0",
```

I think this is simplier

Comment (mrardon) [inline composer.json]: I would agree (and it is much clearer). Mostly was just following the previous style of specifying. Any objects to using this new version constraint?

File: Controller/DefaultController.php [hunk 3]
```diff
@@ -99,13 +100,13 @@ protected function getResque()
      *
      * @return array
      */
-    private function getShowParameters()
+    private function getShowParameters(Request $request)
```
Comment (mrardon) [inline Controller/DefaultController.php:103]: Does anyone know if this injection is valid for the supported Symfony versions? >2.0 I can't seem to find when it was first introduced.

File: composer.json (hunk 2 above)
Comment (lchristian13) [inline composer.json]: I suggest to use it like ~2.0|~3.0, but in theory it should work the same as reminec's suggestion.

File: Controller/DefaultController.php (hunk 3 above)
Comment (mrardon) [inline Controller/DefaultController.php:103]: Appears this is good on 2.0 branch of Symfony and up.

https://github.com/symfony/symfony/blob/2.0/src/Symfony/Component/HttpKernel/Controller/ControllerResolver.php#L115

**Comment by slimus** [comment]
Any news here?

**Comment by badlamer** [comment]
+1

**Comment by madmis** [comment]
+1

**Comment by madmis** [comment]
There also should be changed path to `bootstrap.php.cache`

resque-bundle/BCC/ResqueBundle/Command/StartWorkerCommand.php:44

```
$env['APP_INCLUDE'] = $this->getContainer()->getParameter('kernel.root_dir').'/bootstrap.php.cache';
```

resque-bundle/BCC/ResqueBundle/Command/StartScheduledWorkerCommand.php:36

```
'APP_INCLUDE' => $this->getContainer()->getParameter('kernel.root_dir').'/bootstrap.php.cache',
```

now it's in the `var` directory

**Comment by mrardon** [comment]
Alright so I don't think there is a parameter for the 'var' dir so I'm thinking it would just be something like: 
`$this->getContainer()->getParameter('kernel.root_dir').'/../var/bootstrap.php.cache'`

Whatcha think?

**Comment by madmis** [comment]
I'm agree with you. This path is not configurable in Symfony, so i think it's can be hardcoded.

But, for Symfony2 this path should be different.

Maybe you can use Symfony `VERSION` constant - to define which path using.
Or make it configurable (but i think use VERSION is better)

**Comment by mrardon** [comment]
Yeh I agree there will need to be some logic based on Symfony version.

**Comment by madmis** [comment]
yep

**Comment by mrardon** [comment]
Give that update a try

**Comment by madmis** [comment]
i cheked on symfony 3.0.4 - it's worked. 
But now i can't check symfony2. I will check later.

**Comment by madmis** [comment]
I've tried this fix on new installed Symfony 2.8 - worker worked fine.

**Comment by mrardon** [comment]
So this works on fresh installs of SF3 and SF2 but not upgraded 2=>3 as technically the var directory structure does not have to be changed. Any idea how to properly check for that? To get it to work in my upgraded application I can use:

```
if (Kernel::MAJOR_VERSION == 3 && is_file($bootstrapLocation.'/../var/bootstrap.php.cache')) {
    $bootstrapLocation = $bootstrapLocation.'/../var';
}
```

Not sure if this is ideal though.

**Comment by mpclarkson** [comment]
As it is clear that this bundle is no longer being actively maintained, I have taken it upon myself to maintain a fork here https://github.com/mpclarkson/resque-bundle. I welcome any contributions.

**Pull request closed without merging**